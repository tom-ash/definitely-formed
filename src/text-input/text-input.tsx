import React, { ChangeEvent } from "react";

interface TextInputInterface {
  (props: {
    label: string | React.ReactElement;
    value?: string;
    onChange?(newValue: string, changeEvent?: ChangeEvent): void;
  }): React.ReactElement;
}

export const TextInput: TextInputInterface = props => {
  const { label, value, onChange } = props;

  return (
    <div>
      <label>{label}</label>
      <input
        type="text"
        value={value}
        onChange={onChange ? (e) => onChange(e.target.value, e) : undefined}
      />
    </div>
  );
};
