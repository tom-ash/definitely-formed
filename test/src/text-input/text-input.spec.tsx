import React from "react";
import { describe, expect, test } from "@jest/globals";
import { TextInput } from "../../../src/text-input/text-input";
import renderer from "react-test-renderer";

describe("Text Input", () => {
  describe("label", () => {
    describe("when the provided label is string", () => {
      test("renders with the provided label", () => {
        const component = renderer.create(<TextInput label={"Test"} />);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe("when the provided label is ReactElement", () => {
      test("renders with the provided label", () => {
        const component = renderer.create(<TextInput label={<div>Test</div>} />);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
