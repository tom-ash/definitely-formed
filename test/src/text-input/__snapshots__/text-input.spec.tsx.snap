// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input label when the provided label is ReactElement renders with the provided label 1`] = `
<div>
  <label>
    <div>
      Test
    </div>
  </label>
  <input
    type="text"
  />
</div>
`;

exports[`Text Input label when the provided label is string renders with the provided label 1`] = `
<div>
  <label>
    Test
  </label>
  <input
    type="text"
  />
</div>
`;
